### DISCUSSION TITLE
API Key for Enterprise account

### DISCUSSION THREAD

**Discussion by stigi** [fastlane/fastlane#18049] [Q&A]
Hello,

Apple will be enforcing two-factor auth from February on and we're migrating our CI scripts to using [app_store_connect_api_key](https://docs.fastlane.tools/actions/app_store_connect_api_key/).

Unfortunately it seems like there's no way to create an API Key for for Enterprise accounts. 
I even reached out to our Apple developer advocate and they confirmed:
> At the present time this is not possible, however I will ensure I provide this feedback to the relevant team.

Still `app_store_connect_api_key` has an option for `in_house`, so it seems like somebody somehow was able to generate an in house/enterprise API Key.

How? :)

Thanks for any pointer in the right direction.

**Comment by dinsen** [comment]
See the Note here: https://docs.fastlane.tools/app-store-connect-api/#using-an-app-store-connect-api-key

I just think it's prepared for when you can create auth keys for enterprise accounts.

**Comment by Bernice965** [comment]
I was wondering how to requesting access to the App Store Connect API. When I follow the documentation and go to User & Access, I have API Keys tab, only people? I've tried with the enterprise account in which I am an admin and also with an individual account with the role developer. But no luck with both accounts. I was wondering if this is only limited to account holder or what? Or not available for everyone?




[myfordbenefits](https://www.myfordbenefits.us/)

**Reply by dinsen** [reply]
Hi

Only Admin role can see the Keys tab and only the Account Holder can Request Access to use the App Store Connect API.

Enterprise account can't create API Keys currently.

**Comment by bhat-ganesh** [comment]
Customary 2-year bump 👍🏼 for this.. no hope from Apple 😢.

How is everyone managing without this? 🤔

**Reply by dinsen** [reply]
You can tell Apple you need it here: https://feedbackassistant.apple.com

**Comment by gmogames360** [comment]
Guess this is still a thing in 2024. No way to setup our CI process with enterprise unfortunately

**Reply by dinsen** [reply]
Nope, you need to poke Apple: https://feedbackassistant.apple.com/

**Comment by nevil** [comment]
There's now an Enterprise Program API
https://developer.apple.com/documentation/EnterpriseProgramAPI

**Reply by nevil** [reply]
Huge announcement on August 16th about this long awaited feature:

https://developer.apple.com/news/site-updates/?id=08162024a

**Reply by Noraldeno** [reply]
This took wayyyy too long.

Now that this is available, can we update fastlane to handle this new enterprise program API? 🙏🏽

**Reply by GevaZeichner** [reply]
I experimented with it today, hard-coding the new endpoint since it closely matches the Connect API. I got Spaceship working and successfully retrieved Enterprise provisioning data (certificates, profiles, bundle IDs) using an enterprise key. Next, I’m planning to explore what it takes to support a broader workflow.
Also looks like there's a new discussion [here](https://github.com/fastlane/fastlane/discussions/22205#discussion-7111665).
